### PR TITLE
Update node.d.ts TLS (SSL); added TLSSocket definitions.

### PR DESCRIPTION
**Pull request by Engineer2B**
Add a new type definition for TLSSocket.
The definitions were filled in by using the [Node.JS documentation for TLS](https://nodejs.org/dist/latest-v4.x/docs/api/tls.html#tls_tlssocket_address).

### CODE REVIEW

File: node/node.d.ts [hunk 1]
```diff
@@ -1791,7 +1791,147 @@ declare module "tls" {
...
+         * The cipher name.
+         */
+        name: string;
+        /**
+         * SSL/TLS protocol version.
+         */
+        version: string;
+    }
+
+    export interface TLSSocket extends stream.Duplex {
```
Comment (vvakame) [inline node/node.d.ts]: https://nodejs.org/dist/latest-v4.x/docs/api/tls.html#tls_class_tls_tlssocket
I think this is class, not interface.

File: node/node.d.ts [hunk 2]
```diff
@@ -1791,7 +1791,147 @@ declare module "tls" {
...
+            valid_from: string;
+            valid_to: string;
+            fingerprint: string;
+            serialNumber: string;
+        };
+        /**
+         * Could be used to speed up handshake establishment when reconnecting to the server.
+         * @returns {any} - ASN.1 encoded TLS session or undefined if none was negotiated.
+         */
+        getSession(): any;
```
Comment (mhegazy) [inline node/node.d.ts]: missing getProtocol: https://nodejs.org/api/tls.html#tls_tlssocket_getprotocol

File: node/node.d.ts [hunk 3]
```diff
@@ -1791,7 +1791,147 @@ declare module "tls" {
...
+        /**
+         * Static boolean value, always true.
+         * May be used to distinguish TLS sockets from regular ones.
+         */
+        encrypted: boolean;
+        /**
+         * Returns an object representing the cipher name and the SSL/TLS protocol version of the current connection.
+         * @returns {CipherNameAndProtocol} - Returns an object representing the cipher name
+         * and the SSL/TLS protocol version of the current connection.
+         */
```
Comment (mhegazy) [inline node/node.d.ts]: missing getEphemeralKeyInfo: https://nodejs.org/api/tls.html#tls_tlssocket_getephemeralkeyinfo

**Comment by mhegazy** [comment]
@Engineer2B feel free to add the missing two APIs in a separate PR.

**Pull request merged**